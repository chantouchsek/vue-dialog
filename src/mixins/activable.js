export default {
  name: 'Activable',
  data () {
    return {
      isActive: false
    }
  },
  watch: {
    isActive (val) {
      if (this._dialogInstance) {
        if (this._dialogInstance.isActive !== undefined) {
          this._dialogInstance.isActive = val
        }
      } else {
        if (this.$parent && this.$parent.isActive !== undefined) {
          this.$parent.isActive = val
        }
      }
    }
  },
  methods: {
    close () {
      this.isActive = false
    }
  }
}
