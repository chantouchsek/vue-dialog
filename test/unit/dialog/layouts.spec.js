import Dialog from '../../../src/dialog'
import Vue from 'vue'
import activable from '../../../src/mixins/activable'
// import DialogView from '../../../src/components/view'
// import { mount } from '../../utils'

describe('dialog layouts', () => {
  beforeAll(() => {
    // Vue.component('dialog-view', DialogView)
    // Dialog.prototype.context = context
  })

  // test('Check default layouts when not sended', async () => {
  //   const dlg = new Dialog({
  //     template: '<p>foo</p>'
  //   })
  //   // expect(dlg._wrapper).toHaveProperty('component')
  //   // await dlg.show()
  //   // expect(dlg.element).toMatchSnapshot()
  //   // // expect(asyncData).toHaveBeenCalledTimes(1)
  //   // // expect(dialog.vmd.$data).toHaveProperty('msg', 'foo')
  //   // expect(dlg.innerHTML).toBe('<p>foo</p>')
  //   // dialog.close()
  // })

  test('check error without component', () => {
    expect(() => {
      new Dialog() // eslint-disable-line no-new
    }).toThrow()
  })

  test('Should work with layout with component property', async () => {
    const layout = {
      template: '<div><dialog-child v-bind="$options.propsData" ref="dialog" /></div>'
    }

    const dialog = new Dialog({
      template: '<p>foo</p>'
    }, {
      layout: { component: layout }
    })
    await dialog.show()
    expect(dialog.element).toMatchSnapshot()
    dialog.close()
  })

  it('Should work with layout', async () => {
    const dialog = new Dialog({
      template: '<p>foo</p>',
      layout: {
        template: '<div :width="getWidth"><dialog-child v-bind="$options.propsData" ref="dialog" /></div>'
      }
    })
    await dialog.show()
    expect(dialog.element).toMatchSnapshot()
    expect(dialog.element.textContent).toBe('foo')
    expect(dialog.vm.isLayout).toBe(true)
    expect(dialog.vm.getWidth).toBe('450px')
    dialog.close()
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  it('Should close when in not persistent and not loading state', async () => {
    const dialog = new Dialog({
      template: '<p>foo</p>',
      layout: {
        template: '<div :width="getWidth"><dialog-child v-bind="$options.propsData" ref="dialog" /></div>'
      }
    })
    await dialog.show({ width: '2el' })
    expect(dialog.element).toMatchSnapshot()
    dialog.vm.dismiss()
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  // it('Should not close when in persistent or loading state', async () => {
  //   const dialog = new Dialog({
  //     template: '<p>foo</p>',
  //     layout: {
  //       template: '<div><slot/></div>'
  //     }
  //   })
  //   await dialog.show()
  //   dialog.loading = true
  //   dialog.vm.dismiss()
  //   dialog.close()
  //   // expect(document.body.innerHTML).toBe('')
  // })

  it('Should not hide when in persistent or loading state', async () => {
    const dialog = new Dialog({
      template: '<p>foo</p>',
      layout: {
        template: '<div><dialog-child v-bind="$options.propsData" ref="dialog" /></div>'
      }
    })
    await dialog.show()
    dialog.vm.$el.remove = undefined
    dialog.close()
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  it('Should close when dialog is activable', async () => {
    const dialog = new Dialog({
      template: '<p>foo</p>',
      mixins: [activable],
      layout: {
        template: '<div><dialog-child v-bind="$options.propsData" ref="dialog" /></div>'
      }
    })
    await dialog.show()
    dialog.vmd.isActive = false
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })
})
