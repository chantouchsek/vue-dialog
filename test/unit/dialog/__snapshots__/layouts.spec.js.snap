// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog layouts Should close when in not persistent and not loading state 1`] = `
<div
  class="dialog-layout"
>
  <p>
    foo
  </p>
</div>
`;

exports[`dialog layouts Should work with layout 1`] = `
<div
  class="dialog-layout"
>
  <p>
    foo
  </p>
</div>
`;

exports[`dialog layouts Should work with layout with component property 1`] = `
<div>
  <p>
    foo
  </p>
</div>
`;
